ui/ui-components/pages/_app.tsx

```diff
@@ -74,15 +79,33 @@ GrouparooNextApp.getInitialProps = async (appContext: AppContext) => {
   };
 
   try {
-    const navigationResponse: Actions.NavigationList = await client.request(
-      "get",
-      `/navigation`
-    );
+    let model: Models.GrouparooModelType | null = null;
+
+    const promises: Promise<any>[] = [
+      client.request<Actions.NavigationList>("get", `/navigation`),
+    ];
+
+    if (modelId) {
+      promises.push(
+        client.request<Actions.ModelView>("get", `/model/${modelId}`)
+      );
+    }
```

Comment:
```suggestion
```

Reply:
@edmundito This was kc's change I was referring to in my earlier review - We're still calling `client.request` twice for both the nav list and model, presumably from an earlier implementation. This `promises` array doesn't seem to be used.

Reply:
@pedroslopez Whoops, I thought I deleted those! I just pushed a change that removes them.